qa/rpc-tests/blockdelay_2.py

```diff
@@ -23,40 +23,15 @@ def setup_chain(self, split=False):
         print("Initializing test directory "+self.options.tmpdir)
         initialize_chain_clean(self.options.tmpdir, 3)
 
-    def setup_network(self, split=False):
-        self.nodes = []
-
+    def setup_nodes(self, split=False):
         self.nodes = start_nodes(3, self.options.tmpdir)
-
-        connect_nodes_bi(self.nodes, 0, 1)
-        connect_nodes_bi(self.nodes, 1, 2)
-        #connect_nodes_bi(self.nodes, 0, 2)
-        self.is_network_split = split
-        self.sync_all()
-
-    def split_network(self):
-        # Split the network of three nodes into nodes 0 and 1/2.
-#        assert not self.is_network_split
-        disconnect_nodes(self.nodes[1], 0)
-        disconnect_nodes(self.nodes[0], 1)
-        #disconnect_nodes(self.nodes[0], 2)
-        #disconnect_nodes(self.nodes[2], 0)
-        self.is_network_split = True
         
-    def split_network_2(self):
-        # Split the network of three nodes into nodes 0, 1 and 2.
-        disconnect_nodes(self.nodes[1], 2)
-        disconnect_nodes(self.nodes[2], 1)
-
-
-    def join_network(self):
+    def join_network(self, id = 0):
```

Comment:
I propose to remove both `join_network` and `join_network_2` from this test; the base class implementation can be used setting properly the `id` parameter (please note `sync_blocks` and `sync_mempools` are currently useless, being called on single-element-set, like `self.nodes[X:X+1]`).
Please replace accordingly calls to `join_network_2` with `join_network` with parameter `id=1` (lines #432, #572).